src/shared/lib/GenericAssayUtils/MutationalSignaturesUtils.tsx

```diff
@@ -163,3 +166,35 @@ export function validateMutationalSignatureRawData(
     // we are expecting contribution and pvalue profiles are in pairs
     return _.every(profileIdsGroupByVersion, ids => ids.length === 2);
 }
+
+export function retrieveMutationalSignatureVersionFromData(
+    SignatureProfiles: any[]
+) {
+    let valueToSet: string = 'v2';
+    let uniqueProfileVersion = _.uniq(
+        SignatureProfiles.map(function(obj) {
+            return _.last(obj.split('_'));
+        })
+    );
+    if (uniqueProfileVersion !== undefined) {
+        // give priority to version 3 over version 2
+        if (
+            uniqueProfileVersion.indexOf('v3') !== -1 &&
+            uniqueProfileVersion.indexOf('v2') !== -1
+        ) {
+            valueToSet = Object.values(MutationalSignaturesVersion)[
+                uniqueProfileVersion.indexOf('v3')
+            ];
+        } else {
+            let versionToSet = Object.values(
+                MutationalSignaturesVersion
+            ).indexOf(
```

Comment:
Maybe replace `indexOf` with `includes`?

Reply:
`indexOf` changed to `includes`